include/gatekeeper_lls.h

```diff
@@ -191,6 +196,29 @@ struct lls_config {
 	uint16_t           tx_queue_back;
 };
 
+/*
+ * Functions for resolving IPv4 addresses to link layer addresses.
+ *
+ * resolve_arp() calls the given callback function once when a
+ * resolution is available.
+ *
+ * hold_arp() calls the given callback function function when a
+ * resolution is available and whenever there is an update.
+ *
+ * put_arp() revokes the subscription. If a block calls either
+ * resolve_arp() or hold_arp(), it should later call put_arp()
+ * when it is no longer interested in the map. Callers of
+ * resolve_arp() may choose to put_arp() on a map after some
+ * timeout if no resolution via the callack function is received.
```

Comment:
Since `put_arp()` can be called at any time -- including during the invocation of the callback registered by `hold_arp()` -- I think we must do it the way described above. In other words, it must be the responsibility of the block to only free the resources once notified that the `put_arp()` has succeeded.